upgrade/upgrade-provider.go

```diff
@@ -173,6 +174,35 @@ func UpgradeProvider(ctx Context, repoOrg, repoName string) error {
 			}).AssignTo(&tfSDKUpgrade),
 		)
 	}
+	if ctx.UpgradePfVersion {
+		discoverSteps = append(discoverSteps,
+			step.F("Planning Plugin Framework Update", func() (string, error) {
+				refs, err := gitRefsOf(ctx, "https://"+modPathWithoutVersion(goMod.Bridge.Path),
+					"tags")
+				if err != nil {
+					return "", err
+				}
+				sortedRefs := refs.sortedLabels(func(a string, b string) bool { return a > b })
+
+				var targetVersion semver.Version
+				for _, r := range sortedRefs {
+					if v := strings.TrimPrefix(r, "refs/tags/pf/"); v != r {
+						if targetVersion.LessThan(semver.MustParse(v)) {
+							targetVersion = *semver.MustParse(v)
+						}
+					}
+				}
+
+				// If our target upgrade version is the same as our current version, we skip the update.
+				if targetVersion.String() == goMod.Pf.Version {
```

Comment:
https://github.com/pulumi/upgrade-provider/pull/125/ introduced a function to do this.
```suggestion
				targetVersion := latestSemverTag("pf/", refs)

				// If our target upgrade version is the same as our current version, we skip the update.
				if targetVersion.Original() == goMod.Pf.Version {
```